vllm/model_executor/layers/vocab_parallel_embedding.py

```diff
@@ -60,31 +67,61 @@ def __init__(self,
         self.num_embeddings_padded = pad_vocab_size(num_embeddings,
                                                     padding_size)
         self.embedding_dim = embedding_dim
+
+        quant_method = None
+        if quant_config is not None:
+            quant_method = quant_config.get_quant_method(self)
+
+        # lm_head may be quantized
+        if quant_method is not None:
+            self.linear_method = quant_method
+        else:
+            self.linear_method = UnquantizedLinearMethod()
+
         if params_dtype is None:
             params_dtype = torch.get_default_dtype()
+
         self.tp_size = get_tensor_model_parallel_world_size()
-        # Divide the weight matrix along the vocaburaly dimension.
+        # Divide the weight matrix along the vocabulary dimension.
         self.vocab_start_index, self.vocab_end_index = (
             vocab_range_from_global_vocab_size(
                 self.num_embeddings_padded, get_tensor_model_parallel_rank(),
                 self.tp_size))
         self.num_embeddings_per_partition = (self.vocab_end_index -
                                              self.vocab_start_index)
-        self.weight = Parameter(
-            torch.empty(self.num_embeddings_per_partition,
-                        self.embedding_dim,
-                        dtype=params_dtype))
-        set_weight_attrs(self.weight, {
-            "parallel_dim": 0,
-            "weight_loader": self.weight_loader
-        })
+
+        self.linear_method.create_weights(self,
+                                          self.embedding_dim,
+                                          [self.num_embeddings_per_partition],
+                                          self.embedding_dim,
+                                          self.num_embeddings_per_partition,
```

Comment:
this should be total_embeddings

Reply:
@robertgshaw2-neuralmagic  output_size fixed to be total: `self.embedding_dim`